File: 2025_03/leetcode2025-03-19.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    vector<vector<int>> findMatrix(vector<int>& nums) {
        unordered_map<int, int> cnt;
        int n = 0;
        for (int &i : nums){
            cnt[i]++;
            n = max(n, cnt[i]);
        }
        vector<vector<int>> res;
        for (int i=0; i<n; i++){
            vector<int> tmp;
            for (auto& kv : cnt){
                if (kv.second > 0){
                    tmp.push_back(kv.first);
                    cnt[kv.first]--;
                }
            }
            res.push_back(tmp);
        }
        return res;
    }
};

int main(){
    Solution s;
    vector<int> nums = {1,3,4,1,2,3,1};
    vector<vector<int>> res = s.findMatrix(nums);
    for (auto&i : res){
        for (auto&j : i){
            cout << j << " ";
        }
        cout << endl;
    }
}

File: leetcode2025/2025_03/leetcode2025-03-19.py
from typing import List
from collections import Counter

class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        res = []
        cnt = Counter(nums)
        n = max(cnt.values())
        for i in range(n):
            tmp = []
            for j in cnt:
                if cnt[j] > 0:
                    tmp.append(j)
                    cnt[j] -= 1
            res.append(tmp)
        return res

if __name__ == "__main__":
    s = Solution()
    nums = [1,3,4,1,2,3,1]
    print(s.findMatrix(nums))
